6_visualize.yml

```diff
@@ -72,45 +67,90 @@ targets:
   watermark_fun:
     command: prep_watermark_fun('6_visualize/in/usgs_logo_black.png')
 
-  gif_tasks:
-    command: create_gif_tasks(
+  #### intro frames - animating the link between gages and sparklines ####
+
+  intro_frame_config:
+    command: viz_config[I(c('width','height'))]
+
+  n_intro_frames:
+    command: viz_config[[I('n_intro_frames')]]
+  intro_gif_tasks:
+    command: create_intro_gif_tasks(
+      n_timesteps = n_intro_frames,
+      folders = visualize_folders)
+
+  6_intro_gif_tasks.yml:
+    command: create_intro_gif_makefile(
+      makefile = target_name,
+      task_plan = intro_gif_tasks,
+      remake_file = I('6_visualize.yml'))
+
+  # Build the timestep-specific functions and png files.
+  # When we like how this is looking, we can set task_names=NULL to build all frames.
+  # If you're trying to iterate on just a few png frames, delete 6_visualize/log/6_intro_gif_tasks.ind,
+  # then you can call scmake('6_visualize/log/6_intro_gif_tasks.ind') repeatedly.
+  # Once the .ind file exists, you'll need to call scmake('6_visualize/log/6_intro_gif_tasks.ind', force=TRUE)
+  #   to iterate on all frames at once.
+  trial_intro_frames:
+    command: c(I(c(
+      'a_00_001',
+      'a_00_005',
+      'a_00_009')))
+  6_visualize/log/6_intro_gif_tasks.ind:
+    command: loop_tasks(
+      task_plan=intro_gif_tasks, task_makefile='6_intro_gif_tasks.yml',
+      task_names=trial_intro_frames,
+      num_tries=I(1), sleep_on_error=I(0))
+
+  #### storm frames - animating the progression of the storm, precip, and stage over time ####
+
+  storm_frame_config:
+    command: viz_config[I(c('width','height'))]
+
+  storm_gif_tasks:
+    command: create_storm_gif_tasks(
       timestep_ind = '2_process/out/timesteps.rds.ind',
-      folders = visualize_folders,
-      storm_cfg = storm_frame_config)
+      folders = visualize_folders)
 
-  6_gif_tasks.yml:
-    command: create_gif_makefile(
+  6_storm_gif_tasks.yml:
+    command: create_storm_gif_makefile(
       makefile = target_name,
-      task_plan = gif_tasks,
+      task_plan = storm_gif_tasks,
       remake_file = I('6_visualize.yml'))
 
   # Build the timestep-specific functions and png files.
   # When we like how this is looking, we can set task_names=NULL to build all frames.
-  # If you're trying to iterate on just a few png frames, delete 6_visualize/log/6_gif_tasks.ind,
-  # then you can call scmake('6_visualize/log/6_gif_tasks.ind') repeatedly.
-  # Once the .ind file exists, you'll need to call scmake('6_visualize/log/6_gif_tasks.ind', force=TRUE)
+  # If you're trying to iterate on just a few png frames, delete 6_visualize/log/6_storm_gif_tasks.ind,
+  # then you can call scmake('6_visualize/log/6_storm_gif_tasks.ind') repeatedly.
+  # Once the .ind file exists, you'll need to call scmake('6_visualize/log/6_storm_gif_tasks.ind', force=TRUE)
   #   to iterate on all frames at once.
-  trial_frames:
+  trial_storm_frames:
```

Comment:
Good. I like this pattern. We could duplicate the pattern to display an interesting "thumbnail" frame for the GIF before it is played (which might show the middle of the storm, which is more interesting than the blank map). 

so:
first frame is 0 delay (there is some way to do this but I haven't figured it out) of a chosen timestep
next n frames are "trial/explanation" frames
next k frames are the storm animated through time. 

Seems like your pattern would support more tack on options like that, or a final explanatory set of frames too, as long as that is handled in the GIF building part.

Reply:
yeah, totally!